DA/submissions/tictactoe.py final copy

```diff
@@ -0,0 +1,124 @@
+        player1wins()
+        'break'
+    else:
+        print(' a     b     c')  
+        print('1  -  |  -  |  - ') 
+        print(' _____|_____|_____')     
+        print('2  -  |  -  |  -')  
+        print(' _____|_____|_____')     
+        print('3  -  |  -  |  -')  
+        print('      |     |     ')
+        #asking for third input and tracking    
+        player24 = input('player2/o pick between A1, A2, A3, B1, B2, B3, C1, C2, C3')
+        
+        print(player24)    
+    if player2+player23+player24==('A1A2A3') or ('A1B1C1') or ('A1B2C3') or ('B1B2B3') or ('C1C2C3') or ('C1B2A3'):
+        player2wins()
+        'break'
+    else:
+        print(' a     b     c')  
+        print('1  -  |  -  |  - ') 
+        print(' _____|_____|_____')     
+        print('2  -  |  -  |  -')  
+        print(' _____|_____|_____')     
+        print('3  -  |  -  |  -')  
+        print('      |     |     ')
+        #asking for fourth input and tracking
+        player14=input('player1/x pick between A1, A2, A3, B1, B2, B3, C1, C2, C3')
+        
+        print(player14)
+    if player1+player12+player13 or player12+player13+player14 or player1+player12+player14 or player1+player13+player14==('A1A2A3') or ('A1B1C1') or ('A1B2C3') or ('B1B2B3') or ('C1C2C3') or ('C1B2A3'):
+        player1wins()
+        'break'
+    else:
+        print(' a     b     c')  
+        print('1  -  |  -  |  - ') 
+        print(' _____|_____|_____')     
+        print('2  -  |  -  |  -')  
+        print(' _____|_____|_____')     
+        print('3  -  |  -  |  -')  
+        print('      |     |     ')
+         #asking for fourth input and tracking
+        player25 = input('player2/o pick between A1, A2, A3, B1, B2, B3, C1, C2, C3')
+        
+        print(player25)
+    if  player2+player23+player24 or player23+player24+player25 or player2+player24+player25 or player2+player24+player25==('A1A2A3') or ('A1B1C1') or ('A1B2C3') or ('B1B2B3') or ('C1C2C3') or ('C1B2A3'):
+        player2wins()
+        'break'
+    else:
+        print(' a     b     c')  
+        print('1  -  |  -  |  - ') 
+        print(' _____|_____|_____')     
+        print('2  -  |  -  |  -')  
+        print(' _____|_____|_____')     
+        print('3  -  |  -  |  -')  
+        print('      |     |     ')
+         #asking for fifth input and tracking
+        player15=input('player2/o pick between A1, A2, A3, B1, B2, B3, C1, C2, C3')
+        print(player15)
+    if player1+player12+player13 or player12+player13+player14 or player1+player12+player14 or player1+player13+player14 or player1+player12+player15 or player1+player13+player15 or player12+player13+player15 or player1+player14+player15 or player12+player14+player15 or player13+player14+player15==('A1A2A3') or ('A1B1C1') or ('A1B2C3') or ('B1B2B3') or ('C1C2C3') or ('C1B2A3'):
+        player1wins()
+        'break'
+    else:
+        #nobody won so printing tie
+        print('tie')             
```

Comment:
So we have an algorithm that works but a lot of repeated code. Is there a mechanism we can use to reduce the code?

Reply:
i shortened it